src/sas/qtgui/Perspectives/Fitting/ConstraintWidget.py

```diff
@@ -434,37 +434,107 @@ def onTabCellEdit(self, row, column):
+                "Inconsistent constraint",
+                msg,
+                QtWidgets.QMessageBox.Ok)
+            self.initializeFitList()
+            return
+        # Then check if the parameter is correctly defined with colons
+        # separating model and parameter name
+        parameter_full_name = constraint_text[:constraint_text.index(
+            "=")].strip()
+        if ":" not in parameter_full_name:
+            msg = ("Incorrect constrained parameter definition. Please use "
+                   "colons to separate model and parameter on the rhs of the "
+                   "definition, e.g. M1:scale")
+            QtWidgets.QMessageBox.critical(
+                self,
+                "Inconsistent constraint",
+                msg,
+                QtWidgets.QMessageBox.Ok)
+            self.initializeFitList()
+            return
+        # We can parse the string
+        new_param = item.data(0)[
+                item.data(0).index(':') + 1:item.data(0).index('=')].strip()
+        new_model = item.data(0)[:item.data(0).index(':')].strip()
+        # Check that the symbol is known so we dont get an unknown tab
+        # All the conditional statements could be grouped in one or
+        # alternatively we could check with expression.py, but we would still
+        # need to do some checks to parse the string
+        symbol_dict = self.parent.parent.perspective(
+            ).getSymbolDictForConstraints()
+        if (new_model + "." + new_param) not in symbol_dict:
+            msg = ("Incorrect constrained parameter definition. Please use "
+                   "a single known parameter in the rhs of the constraint "
+                   "definition, e.g. M1:scale = M1.radius + 2")
+            QtWidgets.QMessageBox.critical(
+                self,
+                "Inconsistent constraint",
+                msg,
+                QtWidgets.QMessageBox.Ok)
+            self.initializeFitList()
+            return
+        new_function = item.data(0)[item.data(0).index('=') + 1:].strip()
+        new_tab = self.available_tabs[new_model]
+        # Make sure we are dealing with fit tabs
+        assert(isinstance(tab, FittingWidget))
+        assert(isinstance(new_tab, FittingWidget))
+        # Now check if the user has redefined the constraint and reapply it
+        if new_function != function or new_model != model or new_param != param:
+            # Apply the new constraint
+            constraint = Constraint(param=new_param, func=new_function,
+                                    value_ex=new_model + "." + new_param)
+            new_tab.addConstraintToRow(constraint=constraint,
+                                       row=tab.getRowFromName(new_param))
+            # If the constraint is valid and we are changing model or
+            # parameter, delete the old constraint
+            if self.constraint_accepted and (new_model != model or
+                                             new_param != param):
```

Comment:
Add parentheses and break before `and`. PEP8 is neutral on whether you should indent by 4 or 8 in this case. I don't remember what pylint wants. I use 8 so that the condition is clearly separated from the body.